cosmwasm/packages/accountant/src/state.rs

```diff
@@ -32,6 +32,46 @@ impl fmt::Display for Kind {
     }
 }
 
+#[cw_serde]
+#[derive(Copy, Default, Eq, Hash, PartialOrd, Ord)]
+#[repr(transparent)]
+pub struct Reason(
+    #[serde(with = "fixed_bytes_as_string_serde")]
```

Comment:
We've used `#[serde(with = "hex")]` in other places for this: https://github.com/wormhole-foundation/wormhole/blob/71188d66d113373a81b88e2e9899d6ae21b7ece8/cosmwasm/packages/accountant/src/state/addr.rs#L12-L19

Can also consider leaving it as a String or BString if we take my suggestion to use "crate::arraystring" above.